packages/nvidia_nat_core/src/nat/profiler/parameter_optimization/optimizer_callbacks.py

```
@@ -0,0 +1,98 @@
+            except Exception:
+                logger.exception("OptimizerCallback %s.on_trial_end failed", type(cb).__name__)
+
+    def get_trial_project_name(self, trial_number: int) -> str | None:
+        """Get a trial-specific OTEL project name from the first callback that supports it."""
+        for cb in self._callbacks:
+            fn = getattr(cb, "get_trial_project_name", None)
+            if fn:
+                try:
+                    return fn(trial_number)
+                except Exception:
+                    pass
```

Ruff (0.15.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)